File: wun-ios/Sources/Wun/Foundation/CardRenderer.swift
// SwiftUI renderer for `:wun/Card`. Optional `:title` headline, then
// children stacked vertically, all inside a padded rounded surface.

import SwiftUI

public enum WunCard {
    public static let render: WunComponent = { props, children in
        let title = props["title"]?.stringValue
        return AnyView(
            VStack(alignment: .leading, spacing: 8) {
                if let t = title {
                    SwiftUI.Text(t).font(.system(size: 18, weight: .semibold))
                }
                ForEach(Array(children.enumerated()), id: \.offset) { _, kid in
                    WunView(kid)
                }
            }
            .padding(16)
            .background(
                RoundedRectangle(cornerRadius: 12)
                    .fill(Color.gray.opacity(0.08))
            )
        )
    }
}

File: wun-android/src/main/kotlin/wun/foundation/CardRenderer.kt
// Compose renderer for `:wun/Card`. Optional :title headline, then
// children stacked vertically, all inside a padded rounded surface.

package wun.foundation

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Card
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.serialization.json.JsonPrimitive
import wun.WunComponent
import wun.WunView

object CardRenderer {
    val render: WunComponent = { props, children ->
        val title = (props["title"] as? JsonPrimitive)?.let {
            if (it.isString) it.content else null
        }
        Card(shape = RoundedCornerShape(12.dp), elevation = 0.dp) {
            Column(
                modifier = Modifier.padding(16.dp),
                verticalArrangement = Arrangement.spacedBy(8.dp),
            ) {
                if (title != null) {
                    Text(title, fontSize = 18.sp, fontWeight = FontWeight.SemiBold)
                }
                children.forEach { kid -> WunView(kid) }
            }
        }
    }
}
